src/txmempool.cpp

```diff
@@ -803,12 +798,39 @@ void CTxMemPool::removeStaleTransactions(const CCoinsViewCache * const pCoinsVie
         }
     }
 
-    // mbtr will be removed if they target outdated CertDataHash
+    // Check if any transaction is now invalid due to a certificate update
     for (auto it = mapSidechains.begin(); it != mapSidechains.end(); it++)
     {
-        if (pCoinsView->GetActiveCertDataHash(it->first) != it->second.mcBtrsCertDataHash)
+        CScCertificateView certView = pCoinsView->GetActiveCertView(it->first);
```

Comment:
We should consider the case when both ScCreation and FT/MBTR are in the mempool.
There is no sense to execute this loop, because the `pCoinsView->GetActiveCertView(it->first);` will return
```
if (pSidechain == nullptr)
        return CScCertificateView();
```
Note: pCoinsView is actually a pcoinstip, not a mempool based view. (see the usage)

So, the checks should be moved into corresponding loops after the statement:
```
if (hasSidechainCreationTx(ft.scId))
                continue;
```
Note: no need to check `ftFee/mbtrFee` correctness if we get into `if (hasSidechainCreationTx(ft.scId))`, because if FT/MBTR was applied to mempool, it is consistent with the ScCreation in the mempool.

Reply:
Done.